### PR TITLE
[CI] [GHA] Fix `openvino.test_utils` imports for Mac and Win Python unittests

### PR DESCRIPTION
**Pull request by akashchi**
### Tickets:
 - *123806*

### CODE REVIEW

File: .github/workflows/mac.yml [hunk 1]
```diff
@@ -623,9 +642,9 @@ jobs:
       - name: MO Python API Tests
         run: |
           python3 -m pip install -r ${{ env.LAYER_TESTS_INSTALL_DIR }}/requirements.txt
-          export PYTHONPATH=${{ env.LAYER_TESTS_INSTALL_DIR }}:$PYTHONPATH
-          # TODO: remove setupvars.sh from here; currently, it's used for 'test_utils' installed in '<package>/python/openvino'
-          source ${INSTALL_DIR}/setupvars.sh
+          
+          # Used for 'test_utils' installed in '<test_package>/tests/python/openvino/test_utils'
+          export PYTHONPATH=${{ env.INSTALL_TEST_DIR }}/tests/python/openvino/test_utils:$PYTHONPATH
```
Comment (akashchi) [inline .github/workflows/mac.yml:647]: Should we used `test_utils` from the test package or from the installed OV wheels?

File: .github/workflows/mac.yml [hunk 2]
```diff
@@ -623,9 +642,9 @@ jobs:
       - name: MO Python API Tests
         run: |
           python3 -m pip install -r ${{ env.LAYER_TESTS_INSTALL_DIR }}/requirements.txt
-          export PYTHONPATH=${{ env.LAYER_TESTS_INSTALL_DIR }}:$PYTHONPATH
-          # TODO: remove setupvars.sh from here; currently, it's used for 'test_utils' installed in '<package>/python/openvino'
-          source ${INSTALL_DIR}/setupvars.sh
+          
+          # Used for 'test_utils' installed in '<test_package>/python/openvino/test_utils'
+          export PYTHONPATH=${{ env.INSTALL_TEST_DIR }}/python/openvino/test_utils:$PYTHONPATH
```
Comment (akashchi) [inline .github/workflows/mac.yml:647]: Should we use `test_utils` from the test package or from the installed OV wheels?

Comment (ilya-lavrenov) [reply .github/workflows/mac.yml:647]: Installed OV wheels contain product part of OpenVINO and don't contain any test utils.
So, we need to use it from test package

Comment (akashchi) [reply .github/workflows/mac.yml:647]: are `test_utils` even in the wheels?

Comment (ilya-lavrenov) [reply .github/workflows/mac.yml:647]: there is no `test_utils` module in wheel

Comment (akashchi) [reply .github/workflows/mac.yml:647]: My bad, did not update the page after the first comment, thanks for the clarification.

**Comment by akashchi** [comment]
The previously failing tests with `test_utils` imports are passing on [Mac](https://github.com/openvinotoolkit/openvino/actions/runs/6707859776/job/18230208285?pr=20786) and [Win](https://github.com/openvinotoolkit/openvino/actions/runs/6707859775/job/18230115981). I've removed triggers for them as they were. This PR contains the changes from the [universal `setup-python` action PR](https://github.com/openvinotoolkit/openvino/pull/20715).

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**